target_actionnetwork/sinks.py

```diff
@@ -140,44 +139,59 @@ def preprocess_record(self, record: dict, context: dict) -> dict:
-        if record:
-            if record.get("error"):
-                raise Exception(record.get("error"))
-
-            lists = record.pop("lists",None)
+        state_updates = {}
+        if not record:
+            return None, True, state_updates
+        
+        error = record.get("error")
+        if error:
+            raise Exception(error)
 
-            response = self.request_api(
-                "POST", endpoint=self.endpoint, request_data=record
-            )
-            res_json = response.json()
+        lists = record.pop("lists", None)
 
-            if lists:
-                email_addresses = record["person"]["email_addresses"]
-                phone_numbers = record["person"]["phone_numbers"]
-                if not email_addresses and not phone_numbers:
-                    raise Exception("Email or Phone Number is required to create outreaches")
-                elif lists:
-                    for l in lists:
-                        if l in self.advocacy_campaigns:
-                            self.create_outreach(self.advocacy_campaigns[l], email_addresses, phone_numbers)
-                        else:
-                            adv_camp_id = self.create_advocacy_campaign(l)
-                            self.create_outreach(adv_camp_id, email_addresses, phone_numbers)
+        response = self.request_api("POST", endpoint=self.endpoint, request_data=record)
+        
+        if not response.ok:
+            raise Exception(response.text)
 
-            id = res_json["_links"]["self"]["href"].split("/")[-1]
-            self.logger.info(f"{self.name} created with id: {id}")
-            return id, True, state_updates
-        
\ No newline at end of file
+        res_json = response.json()
+        record_id = res_json["_links"]["self"]["href"].split("/")[-1]
+
+        if lists:
+            self.get_advocacy_campaigns()
+            email_addresses = record["person"].get("email_addresses", [])
+            phone_numbers = record["person"].get("phone_numbers", [])
+            if not email_addresses and not phone_numbers:
+                raise Exception("Email or Phone Number is required to create outreaches. "
+                                "Error during creation of {} record with id: {}".format(self.name, record_id))
+            for list_name in lists:
+                if list_name in self.advocacy_campaigns:
+                    self.create_outreach(self.advocacy_campaigns[list_name], email_addresses, phone_numbers)
+                else:
+                    adv_camp_id = self.create_advocacy_campaign(list_name)
+                    self.create_outreach(adv_camp_id, email_addresses, phone_numbers)
+
+        self.logger.info("{} created with id: {}".format(self.name, record_id))
+
+        # Update state with the new record ID or any other relevant information
+        state_updates.update({
+            'processed_id': record_id,
+            'success': True
+        })
```

Comment:
I think this is not needed if `record_id` and `success status` are passed on the line below. State handles it automatically.

Reply:
Fixed! Tks!